### ISSUE TITLE
Distribute and Align reference object behaviour not optimal

### ISSUE THREAD

**Issue by vhazali** [nus-fboa2016-PL/PowerPointLabs#36]
For Distribute and Align, the reference object behaviour should be:
### If no object is selected:
- Refer to slide radio button is by default selected
- When user clicks on button, show error dialogue – _need to select at least 1 object_
- If user changes reference to any other than slide, show error dialogue – _need to select at least 2 objects_
### If 1 object is selected:
- Refer to slide radio button is by default selected
- When user clicks on button, apply functions accordingly
- If user changes reference to any other than slide, show error dialogue – _need to select at least 2 objects_
### If 2 or more objects are selected
- Checks if the user has previously made a selection on what to refer to and reverts to that
- Otherwise, refer to first shape radio button is by default selected
- When user clicks on button, apply functions accordingly

**Comment by damithc** [comment]
Does this mean settings can automatically change without the user's knowledge?

**Comment by damithc** [comment]
On a related note, I wonder if settings should be a collapsible panel rather than a pop-up dialog. That way, user can keep settings in view if she wishes.

**Comment by vhazali** [comment]
@damithc yes. I guess a better way would be, for two or more objects selected, it will first check if the user has already made a previous selection and go back to that.

Meaning:
**Case 1**
1. user selects 3 shapes, set the reference to between two shapes
2. user selects 1 shape, reference auto update to slide
3. user selects 2 or more shapes again, the reference will be back to between two shapes (instead of first selected shape)

**Case 2**
1. User selects 3 shapes, reference auto update to first selected shape
2. User selects 1 shape, reference auto update to slide
3. User selects 3 shapes, reference auto update to first selected shape

Regarding collapsible panel, I think it's a good idea, but the settings panel will look very cluttered if there's a lot of options in it (like the current distribute settings)

**Comment by damithc** [comment]
Automatically changing settings can be confusing to users, especially when they can't even see the settings. They might think it's a bug. So we need to be careful with this 'smart default' thing.

**Comment by vhazali** [comment]
@damith I understand what you mean. Should I still go ahead and implement this behavior and we see if it's intuitive for users?

**Comment by damithc** [comment]
Let's fix some sensible defaults and keep smart defaults as a future enhancement, once we got the first version out.

**Comment by vhazali** [comment]
Set the default as requested in documents

Aligns to first selected object by default
Distribute within first two selected objects by default
Adjoin also aligns objects by default

ba1dd93aa09e49af123ee2f4eabd3fdcf5b733a6
